src/odbc/odbc.c

```diff
@@ -4785,11 +4785,27 @@ ODBC_FUNC(SQLColumns, (P(SQLHSTMT,hstmt), PCHARIN(CatalogName,SQLSMALLINT),	/* o
 	WIDE))
 {
 	int retcode;
+	DSTR catalog_name = DSTR_INITIALIZER;
+	SQLSMALLINT catalog_size;
+	const char *proc = NULL;
 
 	ODBC_ENTER_HSTMT;
 
+	if (cbCatalogName == SQL_NTS) {
```

Comment:
reading again this if caller specifies a value > 0 (that is a valid size like `"test"` with len 4) catalog won't be prepended. Why not just remove the if and copy always?

Reply:
Because I didn't know `odbc_get_string_size` existed :) Will change, thanks for the tip.

Reply:
Done, let me know if I missed anything.